CMakeLists.txt

```diff
@@ -1,156 +1,186 @@
+    GIT_TAG "6ee1559715fae9480fcaeb81f24d80a4d1e8c407"
+)
+
+FetchContent_Declare(
+    pybind
+    GIT_REPOSITORY "https://github.com/pybind/pybind11"
+    GIT_TAG "59a2ac2745d8a57ac94c6accced73620d59fb844"
+)
+
+message(STATUS "Loading implot ...")
+FetchContent_MakeAvailable(implot)
+
+message(STATUS "Loading pybind ...")
+FetchContent_MakeAvailable(pybind)
+
+message(STATUS "")
+message(STATUS "All dependencies loaded.")
+message(STATUS "")
+
+# Collect files.
+
+set(SOURCE_FILES
+    extern/imgui/imgui.cpp
+    extern/imgui/imgui_draw.cpp
+    extern/imgui/imgui_demo.cpp
+    extern/imgui/imgui_widgets.cpp
+    extern/imgui/imgui_tables.cpp
+    extern/imgui/backends/imgui_impl_glfw.cpp
+    extern/imgui/backends/imgui_impl_opengl3.cpp
+    extern/imgui/misc/cpp/imgui_stdlib.cpp
+    ${implot_SOURCE_DIR}/implot.cpp
+    ${implot_SOURCE_DIR}/implot_demo.cpp
+    ${implot_SOURCE_DIR}/implot_items.cpp
+    ./src/bindings.cpp
+    ./src/imviz.cpp
+    ./src/input.cpp
+    ./src/file_dialog.cpp
+    ./src/binding_helpers.cpp
+    ./src/bindings_implot.cpp
+    ./src/bindings_imgui.cpp
+    ./src/source_sans_pro.cpp
+    ./src/load_image.cpp
+   )
+
+set(HEADER_FILES 
+    ./src/imviz.hpp
+    ./src/input.hpp
+    ./src/file_dialog.hpp
+    ./src/binding_helpers.hpp
+    ./src/bindings_implot.hpp
+    ./src/bindings_imgui.hpp
+    ./src/source_sans_pro.hpp
+    ./src/load_image.hpp
+    ./src/stb_image.h
+    )
+
+# Builds the python bindings module.
+
+pybind11_add_module(${PY_TARGET_NAME} MODULE ${SOURCE_FILES})
+
+target_include_directories(${PY_TARGET_NAME} SYSTEM PUBLIC
+                           extern/imgui/
+                           ${implot_SOURCE_DIR})
+
+target_include_directories(${PY_TARGET_NAME} PUBLIC src/)
+
+IF (WIN32)
+    target_compile_options(${PY_TARGET_NAME} PUBLIC
+                            -DIMGUI_USER_CONFIG="im_user_config.h"
+                            -Zi   # -g
+                            -O2   # msvc has no -O3
+                            # -Wall
+                            -std:c++17)  # note colon!
+    
+    target_link_libraries(${PY_TARGET_NAME} PUBLIC
+                            ${OPENGL_LIBRARIES}
+                            libglew_static
+                            # stdc++fs  # not needed with msvc
+                            pybind11::module
+                            pybind11::embed
+                            glfw)
+ELSEIF(APPLE)
+    target_compile_options(${PY_TARGET_NAME} PUBLIC
+                            -DIMGUI_USER_CONFIG="im_user_config.h"
+                            -g
+                            -O3
+                            -std=c++17)
+
+    target_link_libraries(${PY_TARGET_NAME} PUBLIC
+                            ${OPENGL_LIBRARIES}
+                            libglew_static
+                            pybind11::module
+                            #pybind11::embed  # issues with conda + ARM (Apple M1), see https://github.com/pybind/pybind11/issues/3081
```

Comment:
Is the embedded python runtime needed at all? Might want to remove on other platforms as well.